infra/scripts/container-scripts/create-db.sh

```diff
@@ -1,39 +1,101 @@
+# This script creates or updates database users with appropriate permissions and passwords in PostgreSQL.
+# It also handles read-only and write users based on the naming convention and enables specified Postgres extensions.
+
 #!/bin/bash
 set -euo pipefail
 shopt -s inherit_errexit
 
-PGPASSWORD=$(node secrets get "$PGPASSWORD_KEY")
-export PGPASSWORD
+function create_or_update_user {
+  local user=$1
+  local password_key=$2
+  local db_name=$3
+  local is_read_only=$4
 
-psql -c "CREATE DATABASE $DB_NAME" || true
+  echo "Configuring user: $user"
 
-set +e
-DB_PASSWORD=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 27)
-set -e
+  # Attempt to fetch existing password
+  set +e # Disable script stopping for this command
+  local password=$(node secrets get "$password_key")
+  local fetch_status=$?
+  set -e # Re-enable script stopping on errors
 
-psql -c "create user $DB_USER with encrypted password '$DB_PASSWORD'" || true
+  local password_generated="false"
+  if [[ $fetch_status -ne 0 || -z "$password" ]]; then
+    echo "Generating new password for $user."
+    # Generate a new password if it doesn't exist in SSM
+    set +e
+    password=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 27)
```

Comment:
We should consider using `/dev/rand` since it is more secure and these are passwords. We might have to also consider the case that there is not enough randomness but I doubt it since this runs on the EKS nodes which should basically always have plenty of things to generate randomness from.

Reply:
never heard of  `/dev/rand` 🤷🏼 do you mean `/dev/random` ?